husarion_ugv_lights/src/animation/moving_image_animation.cpp

```diff
@@ -0,0 +1,328 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "husarion_ugv_lights/animation/moving_image_animation.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <filesystem>
+#include <regex>
+#include <stdexcept>
+#include <string>
+
+#include "yaml-cpp/yaml.h"
+
+#include "ament_index_cpp/get_package_prefix.hpp"
+#include "ament_index_cpp/get_package_share_directory.hpp"
+#include "boost/gil.hpp"
+#include "boost/gil/extension/io/png.hpp"
+#include "boost/gil/extension/numeric/resample.hpp"
+#include "boost/gil/extension/numeric/sampler.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "husarion_ugv_utils/yaml_utils.hpp"
+
+namespace husarion_ugv_lights
+{
+
+void MovingImageAnimation::Initialize(
+  const YAML::Node & animation_description, const std::size_t num_led,
+  const float controller_frequency)
+{
+  Animation::Initialize(animation_description, num_led, controller_frequency);
+
+  const auto image_path = ParseImagePath(
+    husarion_ugv_utils::GetYAMLKeyValue<std::string>(animation_description, "image"));
+
+  try {
+    image_center_offset_ =
+      static_cast<int16_t>(std::stoi(husarion_ugv_utils::GetYAMLKeyValue<std::string>(
+        animation_description, "center_offset")));  // in pixels
```

Comment:
```suggestion
    image_center_offset_ =
      husarion_ugv_utils::GetYAMLKeyValue<std::int16_t>(
        animation_description, "center_offset");  // in pixels
```
GetYAMLKeyValue already does conversion